common/motion-dashboard.cpp

```diff
@@ -0,0 +1,190 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2024 Intel Corporation. All Rights Reserved.
+
+#include "output-model.h"
+#include "motion-dashboard.h"
+
+using namespace rs2;
+using namespace rsutils::string;
+
+motion_dashboard::motion_dashboard( std::string name, enum rs2_stream stream)
+    : stream_dashboard( name, 30 )
+    , last_time( glfwGetTime() )
+    , x_value( 0 )
+    , y_value( 0 )
+    , z_value( 0 )
+    , n_value( 0 )
+{
+    dashboard_update_rate = MAX_FRAME_RATE;
+    stream_type = stream;
+    clear( true );
+}
+
+void motion_dashboard::process_frame( rs2::frame f )
+{
+    write_shared_data(
+        [&]()
+        {
+            if( f && f.is< rs2::motion_frame >()
+                && ( f.as< rs2::motion_frame >() ).get_profile().stream_type() == stream_type )
+            {
+                double ts = glfwGetTime();
+                auto it = frame_to_time.find( f.get_profile().unique_id() );
```

Comment:
Please consider if this map is necessary.

Reply:
Yes, it is an unnecessary check with `it`. I removed `it`.